article-analyzer/src/main/java/com/inventorsoft/homework/utility/JsonUtil.java

```diff
@@ -0,0 +1,43 @@
+package com.inventorsoft.homework.utility;
+
+import com.google.gson.Gson;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import org.springframework.stereotype.Component;
+
+@Component
+public class JsonUtil {
+
+  public <T> List<T> readFromJson(String fileName, Class<T[]> clazz) {
+    Gson gson = new Gson();
```

Comment:
make Gson as a spring bean and inject it in classes, don't create the instance every time the method is called